File: 30 Days of Code/Day 10 Binary Numbers/consecutive.c
#include <stdio.h>

int main()
{
	int number = 0;
	scanf("%d", &number);
	int size = 0;
	int binary[64];

	while (number != 0) {
		binary[size++] = number % 2;
		number /= 2;
	}

	int max = 0;
	for (int i = size-1, count = 0; i >= 0; i--) {
		count = binary[i] == 0 ? 0 : ++count;
		max = count > max ? count : max;
	}

	printf("%d", max);

	return 0;
}

File: 30 Days of Code/Day 10 Binary Numbers/Consecutive.java
import java.util.Scanner;

class Consecutive {
	public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int number = input.nextInt();
        char[] binary = Integer.toBinaryString(number).toCharArray();

        int max = 0;
        for (int i = 0, count = 0; i < binary.length; i++) {
            count = binary[i] == '0' ? 0 : ++count;
            max = count > max ? count : max;
        }
        System.out.println(max);
	}
}